Runtime/Scripts/BaseMeshSync.Textures.cs

```diff
@@ -0,0 +1,210 @@
+                foreach (string textureName in textureNames) {
+                    var mat = materialHolder.material;
+                    if (mat == null || !mat.HasTexture(textureName)) {
+                        continue;
+                    }
+
+                    var tex = mat.GetTexture(textureName);
+
+                    if (tex is RenderTexture renderTarget) {
+                        var texture = new Texture2D(renderTarget.width, renderTarget.height,
+                            UnityEngine.TextureFormat.RGBA32, true);
+                        var activeRenderTarget = RenderTexture.active;
+
+                        RenderTexture.active = renderTarget;
+                        texture.ReadPixels(new Rect(0, 0, renderTarget.width, renderTarget.height), 0, 0);
+                        texture.Apply();
+
+                        RenderTexture.active = activeRenderTarget;
+
+                        var savePath = GetSavePath(mat, textureName);
+                        TextureData.WriteToFile(savePath, texture.EncodeToPNG());
+
+                        // Setting the texture can fail if the asset database has not imported it yet.
+                        // If that happens, save it for later and try again:
+                        if (!SetSerializedTextureForMaterial(mat, textureName)) {
+                            pendingMaterialUpdates.Add(new Tuple<Material, string>(mat, textureName));
+
+                            // Set the texture on the material to null for now so the render texture
+                            // is not saved to file again if this gets called again before it was
+                            // set to the texture in the asset library:
+                            mat.SetTexture(textureName, null);
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Sets the texture on the material if it exists in the AssetDatabase, otherwise it schedules a delayed call to try again when the asset might be imported.
+        /// </summary>
+        /// <param name="mat">Material to set the texture on</param>
+        /// <param name="textureName">Name of the texture in the material</param>
+        /// <returns></returns>
+        private bool SetSerializedTextureForMaterial(Material mat, string textureName) {
+            var savePath = GetSavePath(mat, textureName);
+
+            // LoadAssetAtPath can throw an exception if it's called during a domain backup but we have no way to detect that:
+            try {
+                if (AssetDatabase.LoadAssetAtPath<Texture2D>(savePath) == null) {
+                    EditorApplication.delayCall -= UpdatePendingMaterials;
+                    EditorApplication.delayCall += UpdatePendingMaterials;
+
+                    return false;
+                }
+            }
+            catch (Exception ex) {
+                return false;
+            }
```

Comment:
Would it make sense if the handling was in the caller? Including both the registration to dellayCall and the exception handling

Reply:
`SetSerializedTextureForMaterial` is called from multiple places, I'm trying to avoid code duplication.

Reply:
What would be the value of doing that? Wouldn't that just split `SetSerializedTextureForMaterial` into 2 methods?

Reply:
Would it be possible to catch the specific exception type rather than Exception as this could bury other issues unintentionally, or does AssetDatabase.LoadAssetAtPath actaully throw Exception and we're just screwed?

Reply:
I've changed it to catch the UnityException instead.

Reply:
> What would be the value of doing that? Wouldn't that just split SetSerializedTextureForMaterial into 2 methods?

Would allow us to handle the function execution as a whole. It would also break down the responsibilities.

Reply:
Not sure how useful all this is but I've extracted it to another method.